src/client/datascience/variablesView/variableView.ts

```diff
@@ -51,4 +53,54 @@ export class VariableView extends WebviewViewHost<IVariableViewMapping> implemen
     public get title(): string {
         return 'variableView';
     }
+
+    //tslint:disable-next-line:no-any
+    protected onMessage(message: string, payload: any) {
+        switch (message) {
+            case InteractiveWindowMessages.GetVariablesRequest:
+                this.handleMessage(message, payload, this.requestVariables);
+                break;
+            default:
+                break;
+        }
+
+        super.onMessage(message, payload);
+    }
+
+    // Handle message helper function to specifically handle our message mapping type
+    protected handleMessage<M extends IVariableViewPanelMapping, T extends keyof M>(
+        _message: T,
+        // tslint:disable-next-line:no-any
+        payload: any,
+        handler: (args: M[T]) => void
+    ) {
+        const args = payload as M[T];
+        handler.bind(this)(args);
+    }
+
+    // This is called when the UI side requests new variable data
+    private async requestVariables(args: IJupyterVariablesRequest): Promise<void> {
+        // For now, this just returns a fake variable that we can display in the UI
```

Comment:
This is where part 3 will hook up. Probably with a new NativeNotebookActiveVariables service or something to that effect. Just put in a fake variable for now to test the UI and messages working.

Reply:
Does this work not also apply to the interactive window?

Reply:
I believe you should be able to use IJupyterVariables to handle these requests. It should work with native notebooks too as they have an INotebook under the covers.

Reply:
Yeah, I'd not dug in deep yet, but my assumption would be that I'd be able to use IJupyterVariables pretty directly. Figuring most of the work here is just juggling the refresh when your active NativeNotebook document changes in the IDE.

Reply:
@joyceerhl Not as currently specced. Though I'm starting to think that maybe it's worth considering. I think I mentioned this in another comment, but even if we don't remove the current IW variable view right away we could still have this view added in as an option. Customers might prefer it as they can move / dock it around themselves.

Reply:
Is specced even a word? I feel guilty looking at it, will research...